tests/AbstractPdoEventStoreTest.php

```diff
@@ -486,6 +487,31 @@ public function it_doesnt_double_escape_metadata(): void
 
     /**
      * @test
+     * issue: https://github.com/prooph/pdo-event-store/issues/110
+     */
+    public function it_handles_invalid_json(): void
+    {
+        $this->expectException(JsonException::class);
+
+        $event = UserCreated::with(['name' => ['John', 'Jane']], 1);
+        $event = $event->withAddedMetadata('key', 'value');
+
+        $streamName = new StreamName('Prooph\Model\User');
+        $stream = new Stream($streamName, new ArrayIterator([$event]), ['some' => ['metadata', 'as', 'well']]);
+
+        $this->eventStore->create($stream);
+
+        // TODO: Make json somehow invalid to trigger exception.
```

Comment:
Grab the db connection and update the row with a raw query. This for example would be invalid: `{{`

Reply:
MySQL prevents inserting invalid json

Reply:
only if there is a fuck up with the encoding...

Reply:
maybe we should create a custom connection with wrong encoding to get the test failing?

Reply:
I have no time on the weekend, but I'll finish it on monday.